src/typeahead/index.js

```diff
@@ -96,7 +98,7 @@ var Typeahead = React.createClass({
   },
 
   getOptionsForValue: function(value, options) {
-    if (!SHOULD_SEARCH_VALUE(value)) { return []; }
+    if (this.props.skipBlankSearch && !SHOULD_SEARCH_VALUE(value)) { return []; }
```

Comment:
make SHOULD_SEARCH_VALUE a "private" method on the component instead (e.g., `_shouldSearchValue()`) that encapsulates the props check alongside with the current logic.